tests/data_context/test_data_context_test_yaml_config.py

```diff
@@ -1092,3 +1092,181 @@ def test_golden_path_configured_asset_pandas_datasource_configuration(
 
     # No other usage stats calls detected
     assert mock_emit.call_count == 1
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_golden_path_runtime_data_connector_pandas_datasource_configuration(
+    mock_emit, empty_data_context_stats_enabled, test_df, tmp_path_factory
+):
+    """
+    Tests output of RuntimeDataConnector
+    """
+    base_directory = str(
+        tmp_path_factory.mktemp("test_golden_path_pandas_datasource_configuration")
+    )
+
+    create_files_in_directory(
+        directory=base_directory,
+        file_name_list=[
+            "test_dir_charlie/A/A-1.csv",
+            "test_dir_charlie/A/A-2.csv",
+            "test_dir_charlie/A/A-3.csv",
+        ],
+        file_content_fn=lambda: test_df.to_csv(header=True, index=False),
+    )
+
+    context: DataContext = empty_data_context_stats_enabled
+
+    os.chdir(context.root_directory)
+    import great_expectations as ge
+
+    context = ge.get_context()
+    mock_emit.reset_mock()  # Remove data_context.__init__ call
+
+    yaml_config = f"""
+       class_name: Datasource
+
+       execution_engine:
+           class_name: PandasExecutionEngine
+
+       data_connectors:
+           default_runtime_data_connector_name:
+               class_name: RuntimeDataConnector
+               batch_identifiers:
+                   - default_identifier_name
+       """
+
+    # noinspection PyUnusedLocal
+    report_object = context.test_yaml_config(
+        name="my_directory_datasource",
+        yaml_config=yaml_config,
+        return_mode="report_object",
+    )
+
+    assert report_object["execution_engine"] == {
+        "caching": True,
+        "module_name": "great_expectations.execution_engine.pandas_execution_engine",
+        "class_name": "PandasExecutionEngine",
+        "discard_subset_failing_expectations": False,
+        "boto3_options": {},
+    }
+    assert report_object["data_connectors"]["count"] == 1
+    # checking that note has come back
+    assert (
+        report_object["data_connectors"]["default_runtime_data_connector_name"]["note"]
+        == "RuntimeDataConnector will not have data_asset_names until they are passed in through RuntimeBatchRequest"
+    )
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_golden_path_runtime_data_connector_and_inferred_data_connector_pandas_datasource_configuration(
+    mock_emit, empty_data_context_stats_enabled, test_df, tmp_path_factory
+):
+    """
+    Tests the default configuration that we get for
+    """
```

Comment:
Same comment as the above test docstring. Any additional explanation about what behavior we're testing would be a big help for future users of the test 🙇🏽